### PR TITLE
Function refactoring

### PR DESCRIPTION
**Pull request by bgdfry**
## Purpose

Remove our functions for filtering through data from our components into our container. Make functions more flexible than they currently are.

## Approach

1. Started by actually filtering various items in mapStateToProps, rather than using our existing functions
2. After some research, discovered we can use functions in our containers
3. Recreated and refactored our existing functions and moved them into the userEvents container 
4. Removed actions, files, and functions we were no longer using
5. Did some general code sanitation

### Learning

Where I discovered how to use functions outside of our components: http://stackoverflow.com/questions/34003553/redux-what-is-the-correct-way-to-filter-a-data-array-in-reducer

Further explanation in react/redux docs: https://github.com/reactjs/react-redux/blob/master/docs/api.md#inject-todos-and-todoactioncreators-and-counteractioncreators-together-as-actions

### Open Questions and Pre-Merge TODOs
- If we would like to move these into a separate file (per the examples), let's do it. 
- Otherwise, this is ready to merge.

### CODE REVIEW

**Comment by kylem038** [comment]
Ben this is great - you are a value to this group & to humanity.

File: src/components/Home.js [hunk 1]
```diff
@@ -3,71 +3,28 @@ import { Link } from 'react-router';
...
 goToRepos() {
   this.context.router.push('/repos');
 }
 
 grabUserInfo() {
-  fetch(`https://api.github.com/users/bcgodfrey91/events?page=0&callback`, {
+  fetch(`https://api.github.com/users/kylem038/events?page=0&callback`, {
     method: 'GET'
   })
   .then((res) => {return res.json(); })
```
Comment (blakeworsley) [inline src/components/Home.js:23]: I think we should name the responses the same thing.

File: src/containers/userEvents.js [hunk 2]
```diff
@@ -1,45 +1,42 @@
...
+  const listOfIssues = issues.filter((obj) => obj.payload.action===`${issueType}`);
+  return listOfIssues
+}
+
+const getAllCommits = (events) => {
+  const pushEvents = events.filter((ghEvent) => ghEvent.type==='PushEvent')
+  const allCommits = pushEvents.map((obj) => obj.payload.commits.length);
+  return allCommits
+}
+
```
Comment (blakeworsley) [inline src/containers/userEvents.js:22]: Good job moving this all into here! Looks sharp.

**Comment by blakeworsley** [comment]
Looks good Ben! I just had one question, but regardless, it is good to merge.

File: src/components/Home.js (hunk 1 above)
Comment (bgdfry) [reply src/components/Home.js:23]: We definitely can, I just set it up this way as to not confuse the group by using two different instances of `res` (given that they're returning different values)

Comment (blakeworsley) [reply src/components/Home.js:23]: That makes sense. Let's leave it!

**Pull request merged**